src/Firebase/Auth/ApiClient.php

```diff
@@ -183,6 +184,62 @@ public function unlinkProvider(string $uid, array $providers): ResponseInterface
         ]);
     }
 
+    /**
+     * @param array<array<string|int>> $users The users to import
+     * @param array<array<string|int>> $options Import options. Used to specify how passwords are hashed for example
+     *
+     * @throws AuthException
+     * @throws FirebaseException
+     */
+    public function importUsers(array $users, array $options, ProjectId $projectId): ResponseInterface
+    {
+        // Determine the request body
+        $body = [];
+        if (isset($options['hash'])) {
+            if (isset($options['hash']['algorithm'])) {
+                $body['hashAlgorithm'] = $options['hash']['algorithm'];
+            }
```

Comment:
This and the following lines could be simplified with

```php
$body['hashAlgorithm'] = $options['hash']['algorithm'] ?? null;
```

In the end, `null` values could then be filtered out with

```php
$body = array_filter($body, static function ($value) {
    return $value !== null;
});
```

Reply:
The static keyword before the callback function puzzles me, I'd love to know more about the difference with

```php
$body = array_filter($body, function ($value) {
    return $value !== null;
});
```

Reply:
It's an optimization for lambdas that don't need to reference `$this`. With `static function`, the PHP engine doesn't need to make `$this` available to the function and later garbage collect it.